classes/view_submissions.php

```diff
@@ -1308,23 +1308,8 @@ private function prevent_direct_user_input($confirm) {
         $candeleteownsubmissions = has_capability('mod/surveypro:deleteownsubmissions', $this->context);
         $candeleteotherssubmissions = has_capability('mod/surveypro:deleteotherssubmissions', $this->context);
         $cansavesubmissiontopdf = has_capability('mod/surveypro:savesubmissiontopdf', $this->context);
-
-        if ($this->action == SURVEYPRO_NOACTION) {
-            return true;
-        }
-        if ($confirm == SURVEYPRO_ACTION_EXECUTED) {
-            return true;
-        }
-        if ($confirm == SURVEYPRO_CONFIRMED_NO) {
-            return true;
-        }
-
-        $canseeotherssubmissions = has_capability('mod/surveypro:seeotherssubmissions', $this->context);
```

Comment:
Maybe you saw this line of code. But this one belong to a bunch of duplicate rows.
The variable $canseeotherssubmissions is still alive in:
https://github.com/kordan/moodle-mod_surveypro/blob/relevant_fix/classes/view_submissions.php#L1305

Reply:
Aha, it could be that I did not see the dupes. I remember that I looked for uses of `canduplicateotherssubmissions` and when I saw that it was being used is when I reported to you. Sure I missed it was declared twice.

So, all clear now, thanks!